### ISSUE TITLE
False report for RS1024 when SymbolEqualityComparer is used

### ISSUE THREAD

**Issue by matkoch** [dotnet/roslyn-analyzers#4412]
### Analyzer

**Diagnostic ID**: RS1024

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.100-rc.2](https://dotnet.microsoft.com/download/dotnet/5.0)

### Describe the bug

Analyzer doesn't take passed `equalityComparer` into account.

### Steps To Reproduce

```
var asyncMethods = type.GetMembers()
    .OfType<IMethodSymbol>()
    .Where(x => x.IsAsync)
    .ToLookup(x => x.ContainingType, x => x, SymbolEqualityComparer.Default) // RS1024
```

### Expected behavior

No warning.

### Actual behavior

Warning.

**Comment by yfital** [comment]
Another one for the bunch on .netcore5 final 5.0.100

            ImplementedInterfaces = typeSymbol.AllInterfaces
                .Distinct<INamedTypeSymbol>(SymbolEqualityComparer.Default)
                .ToList();

**Comment by Evangelink** [comment]
I can reproduce the error, this issue also relates to #4469
